src/Unosquare.Swan/Components/ObjectMapper.cs

```diff
@@ -107,152 +149,114 @@ public TDestination Map<TDestination>(object source, bool autoResolve = true)
-        /// <param name="destinationProperty">The destination property.</param>
-        /// <param name="sourceProperty">The source property.</param>
-        /// <returns>
-        /// An object map representation of type of the destination property 
-        /// and type of the source property
-        /// </returns>
-        public ObjectMap<TSource, TDestination> MapProperty
-            <TDestinationProperty, TSourceProperty>(
-                Expression<Func<TDestination, TDestinationProperty>> destinationProperty,
-                Expression<Func<TSource, TSourceProperty>> sourceProperty)
-        {
-            var memberDestinationExpression = destinationProperty?.Body as MemberExpression;
-            var propertyDestinationInfo = memberDestinationExpression?.Member as PropertyInfo;
-
-            if (propertyDestinationInfo == null)
+            // Copy source properties
+            foreach (var sourceProperty in filteredSourceProperties)
             {
-                throw new Exception("Invalid destination expression");
-            }
+                var targetProperty = targetProperties
+                    .First(s => s.Name.ToLowerInvariant() == sourceProperty.Key);
 
-            var sourceMembers = new List<PropertyInfo>();
-            var initialExpression = sourceProperty?.Body as MemberExpression;
+                if (requiredProperties != null && !requiredProperties.Contains(targetProperty.Name.ToLowerInvariant()))
+                    continue;
 
-            while (true)
-            {
-                var propertySourceInfo = initialExpression?.Member as PropertyInfo;
+                if (ignoredProperties != null && ignoredProperties.Contains(targetProperty.Name.ToLowerInvariant()))
+                    continue;
 
-                if (propertySourceInfo == null) break;
-                sourceMembers.Add(propertySourceInfo);
-                initialExpression = initialExpression.Expression as MemberExpression;
+                try
+                {
+                    // Direct Copy
+                    if (targetProperty.PropertyType == sourceProperty.Value.Type)
+                    {
+                        if (sourceProperty.Value.Type.GetTypeInfo().IsEnum)
+                        {
+                            targetProperty.SetValue(target,
+                                Enum.ToObject(targetProperty.PropertyType, sourceProperty.Value.Value));
+                            continue;
+                        }
+
+                        targetProperty.SetValue(target, sourceProperty.Value.Value);
+                        copiedProperties++;
+                        continue;
+                    }
+
+                    // String to target type conversion
+                    if (Definitions.BasicTypesInfo[targetProperty.PropertyType]
+                        .TryParse(sourceProperty.Value.Value.ToStringInvariant(), out var targetValue))
```

Comment:
maybe assign sourceProperty.Value.Value and Type to local variables. They're being repeated through all the method....